File: solutions/cpp/0045.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int ret = 0;
        int end = 0;
        int farthest = 0;

        for (int i = 0; i < nums.size() - 1; i++) {
            farthest = max(farthest, i + nums[i]);
            if (i == end) {
                ret++;
                end = farthest;
            }
        }

        return ret;
    }
};

File: solutions/python3/0045.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        ret = 0
        end = 0
        farthest = 0

        for i in range(len(nums) - 1):
            farthest = max(farthest, i + nums[i])
            if i == end:
                ret += 1
                end = farthest

        return ret